flutter_appauth/android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java

```diff
@@ -464,22 +463,43 @@ private void finishWithSuccess(Object data) {
         }
     }
 
-    private void finishWithError(String errorCode, String errorMessage, String errorDetails) {
+    private void finishWithError(String errorCode, String errorMessage, @Nullable Exception cause) {
         if (pendingOperation != null) {
-            pendingOperation.result.error(errorCode, errorMessage, errorDetails);
+            pendingOperation.result.error(errorCode, errorMessage, createErrorMap(cause));
             pendingOperation = null;
         }
     }
 
     private void finishWithDiscoveryError(AuthorizationException ex) {
-        finishWithError(DISCOVERY_ERROR_CODE, String.format(DISCOVERY_ERROR_MESSAGE_FORMAT, ex.error, ex.errorDescription), getCauseFromException(ex));
+        finishWithError(DISCOVERY_ERROR_CODE, String.format(DISCOVERY_ERROR_MESSAGE_FORMAT, ex.error, ex.errorDescription), ex);
     }
 
     private void finishWithEndSessionError(AuthorizationException ex) {
-        finishWithError(END_SESSION_ERROR_CODE, String.format(END_SESSION_ERROR_MESSAGE_FORMAT, ex.error, ex.errorDescription), getCauseFromException(ex));
+        finishWithError(END_SESSION_ERROR_CODE, String.format(END_SESSION_ERROR_MESSAGE_FORMAT, ex.error, ex.errorDescription), ex);
+    }
+
+    private Map<String, Object> createErrorMap(@Nullable Exception cause) {
+        @Nullable AuthorizationException authException = cause instanceof AuthorizationException ? (AuthorizationException) cause : null;
+        Map<String, Object> responseMap = new HashMap<>();
+        responseMap.put("legacy_error_details", getCauseFromException(cause));
+
+        if (authException != null) {
+            boolean userDidCancel = authException.equals(AuthorizationException.GeneralErrors.USER_CANCELED_AUTH_FLOW);
+            responseMap.put("type", String.valueOf(authException.type));
+            responseMap.put("code", String.valueOf(authException.code));
+            responseMap.put("error", authException.error);
+            responseMap.put("error_description", authException.errorDescription);
+            responseMap.put("error_uri", authException.errorUri == null ? null : authException.errorUri.toString());
+            responseMap.put("root_cause_error", authException.getCause() == null ? "" : authException.getCause().toString());
```

Comment:
Can this be made to return a null instead of an empty string when the cause is null?

Reply:
Done :)